system/ee/ExpressionEngine/Addons/channel/mod.channel.php

```diff
@@ -3986,9 +3986,13 @@ public function category_heading()
 
         // Is the category being specified by name?
 
-        if ($qstring != '' and $this->reserved_cat_segment != '' and in_array($this->reserved_cat_segment, explode("/", $qstring)) and ee()->TMPL->fetch_param('channel')) {
-            $qstring = preg_replace("/(.*?)\/" . preg_quote($this->reserved_cat_segment) . "\//i", '', '/' . $qstring);
-
+        if (
+            (
+                ($this->reserved_cat_segment != '' and in_array($this->reserved_cat_segment, explode("/", $qstring)))
+                OR ee()->TMPL->fetch_param('category_url_name')
```

Comment:
Can we use `category_url_title` here instead? `category_url_name` is not used anywhere else in EE

Reply:
Sure! I thought I already did it! I'm sorry!